x/csr/types/genesis.go

```diff
@@ -15,9 +19,41 @@ func DefaultGenesis() *GenesisState {
 }
 
 // Validate performs basic genesis state validation returning an error upon any
-// failure.
+// failure. Validation will check that there are no duplicate CSRPools registered,
+// that there are not the same NFTs registered for different groups, and that the smart contracts
+// in a given pool are not repeated elsewhere
 func (gs GenesisState) Validate() error {
-    // this line is used by starport scaffolding # genesis/types/validate
+	seenCSRPool := make(map[string]bool)
+	seenCSRNFTs := make(map[string]bool)
+
+	for _, csr := range gs.Csrs {
+		// Validate that the csr object was correctly formatted
+		if err := csr.Validate(); err != nil {
+			return err
+		}
+
+		pool := csr.CsrPool
+		// Validate that the pool is correctly formatted
+		if err := pool.Validate(); err != nil {
```

Comment:
Can just do csr.CsrPool.Validate, no need to initialize another variable

Reply:
i use the pool address later on